lib/src/profile/profile_view.dart

```diff
@@ -0,0 +1,148 @@
+import 'package:bedrock_flutter/src/auth/user_model.dart';
+import 'package:bedrock_flutter/src/profile/profile_controller.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+class ProfileView extends StatefulWidget {
+  static const String appBarLabel = 'Profile';
+
+  const ProfileView({Key? key}) : super(key: key);
+
+  @override
+  State<ProfileView> createState() => _ProfileViewState();
+}
+
+class _ProfileViewState extends State<ProfileView> {
+  final _formKey = GlobalKey<FormState>();
+  late String _email;
+  late String _firstName;
+  late String _lastName;
+
+  void _submit() async {
+    final controller = Provider.of<ProfileController>(context, listen: false);
+    if (_formKey.currentState!.validate()) {
+      controller.updateUser(
+        User(
+          email: _email,
+          firstName: _firstName,
+          lastName: _lastName,
+        ),
+      );
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final controller = Provider.of<ProfileController>(context);
+
+    return FutureBuilder(
+      future: controller.user,
+      builder: (context, AsyncSnapshot<User?> snapshot) {
+        if (snapshot.hasData) {
+          switch (snapshot.connectionState) {
+            case ConnectionState.none:
+              return const CircularProgressIndicator.adaptive();
+            case ConnectionState.waiting:
+              return const CircularProgressIndicator.adaptive();
+            case ConnectionState.active:
+              return const CircularProgressIndicator.adaptive();
```

Comment:
You can use the syntax like this
```
          case ConnectionState.none:
          case ConnectionState.waiting:
          case ConnectionState.active:
              return const CircularProgressIndicator.adaptive();
```

Reply:
I didn't know this was possible in Dart. Awesome!